provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
On 22 June 2016 at 12:00, Fabian Reinartz <notifications@github.com> wrote:

> I'm just trying to convey that, other than govet, golint is not right by
> default but merely provides suggestions. The noisiness is even tweak-able
> via configuration flag.
>
That's not disputed. But in this case, nothing is wrong with golint. It
merely implements what's clearly written in the style guide:
https://github.com/golang/go/wiki/CodeReviewComments#doc-comments .
If you have no intention to change the style guide, then there is indeed
nothing to debate.

-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales with
Company No. 6343600 | Local Branch Office | AG Charlottenburg  | HRB 110657B